File: 03-java/Game.java
import java.util.Random;

public class Game {
    public static final int MIN_NUMBER = 1;
    public static final int MAX_NUMBER = 100;

    private final Random random;
    private int secretNumber;
    private int attempts;

    public Game() {
        random = new Random();
        reiniciar();
    }

    public String comprobarNumero(int number) {
        if (number < MIN_NUMBER || number > MAX_NUMBER) {
            throw new IllegalArgumentException("El número debe estar entre 1 y 100.");
        }

        attempts++;

        if (number < secretNumber) {
            return "Muy bajo";
        }

        if (number > secretNumber) {
            return "Muy alto";
        }

        return "¡Correcto!";
    }

    public int obtenerIntentos() {
        return attempts;
    }

    public void reiniciar() {
        secretNumber = random.nextInt(MAX_NUMBER - MIN_NUMBER + 1) + MIN_NUMBER;
        attempts = 0;
    }
}

File: 05-kotlin/Game.kt
import java.util.Random

/** Mantiene la lógica y el estado de una partida. */
class Game {
    companion object {
        const val MIN_NUMBER = 1
        const val MAX_NUMBER = 100
    }

    private val random = Random()
    private var secretNumber = 0

    var intentos = 0
        private set

    init {
        reiniciar()
    }

    fun comprobarNumero(number: Int): String {
        require(number in MIN_NUMBER..MAX_NUMBER) {
            "El número debe estar entre 1 y 100."
        }

        intentos++

        return when {
            number < secretNumber -> "Muy bajo"
            number > secretNumber -> "Muy alto"
            else -> "¡Correcto!"
        }
    }

    fun reiniciar() {
        secretNumber = random.nextInt(MAX_NUMBER - MIN_NUMBER + 1) + MIN_NUMBER
        intentos = 0
    }
}
